test/extended/util/test.go

```diff
@@ -81,6 +83,11 @@ func InitTest(dryRun bool) error {
 			return err
 		}
 		TestContext.Host = cfg.Host
+		c, err := kclientset.NewForConfig(cfg)
+		if err != nil {
+			klog.Fatal("Error loading client: ", err)
+		}
+		TestContext.IPFamily = getDefaultClusterIPFamily(c)
```

Comment:
We don't need to call `getDefaultClusterIPFamily()`; we already know whether the cluster is ipv4/ipv6/dual-stack based on reading the OCP network config. You can set `context.IPFamily` in `initializeTestFramework` in `cmd/openshift-tests/provider.go` based on `config.HasIPv4`/`config.HasIPv6`.

Reply:
Cool, this will avoid the need of hitting the apis since we are basing this on config.

Reply:
Done, I am not totally sure the way I used hasIPv4 / v6 has the same semantic as checking the ipfamily of the primary clusterip (especially in case of a dual stack cluster).